libs/libc/misc/lib_mutex.c

```diff
@@ -329,15 +356,9 @@ int nxmutex_unlock(FAR mutex_t *mutex)
 #if defined(CONFIG_BUILD_FLAT) || defined(__KERNEL__)
 int nxmutex_reset(FAR mutex_t *mutex)
 {
-  int ret;
+  mutex->holder = NXMUTEX_RESET;
 
-  ret = nxsem_reset(&mutex->sem, 1);
-  if (ret >= 0)
-    {
-      mutex->holder = NXMUTEX_NO_HOLDER;
-    }
-
-  return ret;
+  return nxsem_reset(&mutex->sem, 1);
```

Comment:
if we don't handle nxsem_reset error, it's better  change the return type of nxmutex_reset to void

Reply:
Done